fetch/cosmwasm_contract/src/migration.rs

```diff
@@ -0,0 +1,28 @@
+use crate::access_control::wipe_all_roles_from_storage;
+use crate::contract::initialise_contract_state;
+use crate::error::ERR_ALREADY_REFUNDED;
+use crate::msg::MigrateMsg;
+use crate::state::is_state_valid;
+use cosmwasm_std::{entry_point, DepsMut, Env, Response, StdError, StdResult};
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, env: Env, msg: MigrateMsg) -> StdResult<Response> {
```

Comment:
For some reason you can't have `MessageInfo` which contains sender and sent funds during migration